### ISSUE TITLE
Narrator: Dialog control reads out meta information about header text

### ISSUE THREAD

**Issue by lilymatcha** [OfficeDev/office-ui-fabric-react#11286]
### Environment Information

- **Package version(s)**: 6.131.0
- **Browser and OS versions**: Edge latest, Windows 10 latest

### Describe the issue:

When you turn on Narrator and open up a Dialog, even on the [example page](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog), it reads out not only the text, but also the text type it is (such as "header level 2").

For example, in the first demo on the page, if you try to read the header for the dialog with Narrator, it should read, "Missing subject." However, it actually reads out, "Missing header level 2 subject."

### Please provide a reproduction of the issue in a codepen:

Actually, this reproes on the Dialog [example page](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog). It does not repro on the [Modal example](https://developer.microsoft.com/en-us/fabric#/controls/web/modal), which is interesting given that Dialog is based off of Modal.

#### Actual behavior:

Narrator reads out "Missing header level 2 subject."

#### Expected behavior:

Narrator reads out "Missing subject."

### Documentation describing expected behavior

"Type of DOM node that represents the textual content of an attribute or an element. A Text node has no child nodes." ([Source](https://www.w3.org/TR/accname-1.1/#dfn-text-node)) While this doesn't explicitly say that the text metadata shouldn't be narrated, I feel like it is implied that this is not expected behavior, especially given that the metadata is inserted in between words in the header.

**Comment by lilymatcha** [comment]
After doing more research, I see that I misunderstood the spec, and that reading out the "heading level 2" is in fact expected behavior ([source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role)).

However, the fact that it gets read out in between words in the header seems like unexpected behavior to me (ex. rather than "Missing header level 2 subject," it should be, "Missing subject header level 2" or "Header level 2 missing subject"). I'll leave the bug open as a means of flagging that issue.

**Comment by aneeshack4** [comment]
@lilymatcha Thanks for reaching out to us! This is the right behavior - AFAIK, we can't change the order of words in Narrator's reading out. If you know how to, please lmk.
